packages/block-editor/src/hooks/utils.js

```diff
@@ -19,3 +28,23 @@ export const cleanEmptyObject = ( object ) => {
 	);
 	return isEmpty( cleanedNestedObjects ) ? undefined : cleanedNestedObjects;
 };
+
+export function transformStyles(
+	activeSupports,
+	migrationPaths,
+	result,
+	source
+) {
+	const firstBlockAttributes = source[ 0 ]?.attributes;
+	forEach( activeSupports, ( isActive, support ) => {
+		if ( isActive ) {
+			migrationPaths[ support ].forEach( ( path ) => {
+				const styleValue = get( firstBlockAttributes, path );
+				if ( styleValue ) {
+					set( result.attributes, path, styleValue );
```

Comment:
In any similar approach it would be better not to mutate the object.

Reply:
Fixed 👍